service/src/main/java/greencity/service/EventServiceImpl.java

```diff
@@ -0,0 +1,274 @@
+import greencity.constant.ErrorMessage;
+import greencity.dto.event.EventCreationDtoRequest;
+import greencity.dto.event.EventDto;
+import greencity.dto.event.EventEditDto;
+import greencity.dto.event.EventSendEmailDto;
+import greencity.dto.user.PlaceAuthorDto;
+import greencity.dto.user.UserVO;
+import greencity.entity.Event;
+import greencity.entity.EventDayDetails;
+import greencity.entity.EventImage;
+import greencity.entity.User;
+import greencity.exception.exceptions.EventNotFoundException;
+import greencity.exception.exceptions.NotFoundException;
+import greencity.exception.exceptions.NotSavedException;
+import greencity.repository.EventDayDetailsRepo;
+import greencity.repository.EventRepo;
+import greencity.repository.UserRepo;
+import jakarta.servlet.http.HttpServletRequest;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.modelmapper.ModelMapper;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.time.LocalTime;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static greencity.constant.AppConstant.AUTHORIZATION;
+import static greencity.constant.AppConstant.DEFAULT_EVENT_IMAGE;
+
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class EventServiceImpl implements EventService {
+    private final EventRepo eventRepo;
+    private final UserRepo userRepo;
+    private final FileService fileService;
+    private final UserService userService;
+    private final RestClient restClient;
+    private final HttpServletRequest httpServletRequest;
+    private final ModelMapper modelMapper;
+    private final EventDayDetailsRepo eventDayDetailsRepo;
+
+
+    @Override
+    @Transactional
+    public EventDto saveEvent(EventCreationDtoRequest eventCreationDtoRequest,
+                              List<MultipartFile> images,
+                              String userEmail) {
+
+        // Find the user by email
+        User user = userRepo.findByEmail(userEmail)
+                .orElseThrow(() -> new NotFoundException(ErrorMessage.USER_NOT_FOUND_BY_EMAIL + userEmail));
+
+        // Map eventCreationDtoRequest to Event entity using EventCreationDtoMapper
+        Event eventToSave = modelMapper.map(eventCreationDtoRequest, Event.class);
+        eventToSave.setAuthor(user);
+
+        // Attempt to save the event
+        try {
+            Event savedEvent = eventRepo.save(eventToSave);
+
+            // Handle images after event is saved
+            List<EventImage> eventImages = uploadImages(images);
+            savedEvent.setImages(eventImages);
+
+            // Re-save the event with images
+            eventRepo.save(savedEvent);
```

Comment:
I do not understand something, why first save the event, then download the image and save the event again. Since there is an annotation, if something goes wrong with the loading of the image, it will still not save the event

Reply:
We will improve this method later. But now, this method works correctly. The logic is like this: First, we save the event to get its id. Then, we use this id to create a record in the EventParticipant table. After that, we handle the images and add their id to the Event.